department-of-veterans-affairs/va.gov-cms
# VACMS-3810: Fixes GTM dataLayer issues and adds tests.

## Description

See #3810.  In the course of fixing another bug, a variable scope was stomped on and created a new, worse bug 🤦 

## Testing done
Tested manually and confirmed fix, added Behat tests to protect against regressions.

## Screenshots
Before:
<img width="442" alt="Screen Shot 2020-12-17 at 12 41 35 PM" src="https://user-images.githubusercontent.com/1318579/102525747-b1e0ac80-4068-11eb-9fcf-18e977f6e064.png">

After:
<img width="522" alt="Screen Shot 2020-12-17 at 1 07 14 PM" src="https://user-images.githubusercontent.com/1318579/102525808-cae95d80-4068-11eb-9238-e957d71145d7.png">

## QA steps

As any authenticated user:
1. Visit a [node with a content owner](http://va-gov-cms.lndo.site/coronavirus-veteran-frequently-asked-questions)
1. Open the developer console and view the `dataLayer` variable.
1. Confirm that the dataLayer contains values for fields:
     - [x] `nodeID`
     - [x] `pageTitle`
     - [x] `contentTitle`
     - [x] `contentType`, and 
     - [x] `pagePath`.

### Definition of Done


### Select Team for PR review


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
